test/coverage_specs2_with_junit/expected-coverage.dat

```diff
@@ -9,15 +9,15 @@ FNDA:1,coverage_specs2_with_junit/A1$::a1 (Z)Lcoverage_specs2_with_junit/B1$;
 FNDA:0,coverage_specs2_with_junit/A1::a1 (Z)Lcoverage_specs2_with_junit/B1$;
 FNF:4
 FNH:3
-BRDA:5,0,0,0
-BRDA:5,0,1,1
+BRDA:5,0,0,1
+BRDA:5,0,1,0
 BRF:2
 BRH:1
 DA:3,1
 DA:5,1
-DA:6,1
+DA:6,0
 DA:7,1
-LH:4
+LH:3
```

Comment:
Not sure where the changes in the coverage files comes from but after generating and inspecting coverage reports new changes are definitely more accurate as the failure case for `A1.a1(false)` was actually not covered and is now depicted as a missing branch in the report.

Before : 

![expected-coverage-scalatest](https://github.com/bazelbuild/rules_scala/assets/2460514/e17a6b3c-f5fa-4ae2-b1cc-10db9d52e53c)

After :